### PR TITLE
Monuments-controller

### PR DESCRIPTION
**Pull request by Glod33**

### CODE REVIEW

File: app/controllers/monuments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class MonumentsController < ApplicationController
+  skip_before_action :authenticate_user!, only: :index
+
+  def index
+    if params[:search].present?
+      @monuments = Monuments.global_search(params[:search])
+    else
+      @monuments = Monument.near([user.lat, user.lng], 5)
+    end
```
Comment (wJoenn) [inline app/controllers/monuments_controller.rb:9]: We don't have a search feature yet.
There's no point in adding that now.

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/controllers/monuments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class MonumentsController < ApplicationController
+  skip_before_action :authenticate_user!, only: :index
+
+  def index
+    @monuments = Monument.near([user.lat, user.lng], 5)
+  end
+
+  def show
+    @monument = Monument.find(params[:id])
+  end
```
Comment (wJoenn) [inline app/controllers/monuments_controller.rb:10]: You're not using the policy_scope anymore.
And shouldn't you be authorizing the @monument in the show ?

File: app/models/monument.rb [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-  has_many :architects, through: :architect_monuments
   has_many :histories, dependent: :destroy
   has_one_attached :photo
 
   reverse_geocoded_by :lat, :lng, address: :location
   after_validation :reverse_geocode
 
-  validates :name, :description, :completion_date, :lat, :lng, :style, presence: true
-  # validates :photo, presence: true
+  validates :name, :description, :lat, :lng, :city, :country, :country_code, :photo, presence: true
```
Comment (wJoenn) [inline app/models/monument.rb:8]: I just realized but this is not gonna work without the migrations files.
You can't validate the presence of an attribute that isn't part of the db's table.

Do you really need to edit this model for the rest to work ? 🤔 
If not then you can remove that and it'll be updated it my current PR.
If you do need that then copy the migration files from my PR and add them to yours

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/controllers/monuments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class MonumentsController < ApplicationController
+  skip_before_action :authenticate_user!, only: :index
+
+  def index
+    @monuments = Monument.near([user.lat, user.lng], 5)
+    @monuments = policy_scope(Monument)
```
Comment (wJoenn) [inline app/controllers/monuments_controller.rb:6]: Either one or the other but not both otherwise the value of @monuments is just overwritten

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/models/monument.rb (hunk 3 above)
Comment (Glod33) [reply app/models/monument.rb:8]: I'll delete the file!

File: app/policies/monument_policy.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class MonumentPolicy < ApplicationPolicy
+  class Scope < Scope
+    # NOTE: Be explicit about which records you allow access to!
+    def resolve
+      scope.Monument.near([user.lat, user.lng], 5)
+    end
+
+    def show
+      true
+    end
```
Comment (wJoenn) [inline app/policies/monument_policy.rb:10]: `show` should be `show?` and outside the scope

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/models/monument.rb (hunk 3 above)
Comment (wJoenn) [reply app/models/monument.rb:8]: You haven't yet

File: app/policies/monument_policy.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class MonumentPolicy < ApplicationPolicy
+  class Scope < Scope
+    # NOTE: Be explicit about which records you allow access to!
+    def resolve
+      scope.Monument.near([user.lat, user.lng], 5)
+    end
+  end
```
Comment (wJoenn) [inline app/policies/monument_policy.rb:7]: I asked a TA about this while I we were reviewing my code and he told me that the scope should `scope.all`
Then in the controller you do the near after the scope
`@monuments = policy_scope(Monument).near([user.lat, user.lng], 5)`

**Review by wJoenn — CHANGES_REQUESTED** [review]

**Review by wJoenn — APPROVED** [review]

**Pull request merged**